polyphemus/lib/magma_record_file_etl.rb

```diff
@@ -0,0 +1,119 @@
+require_relative "magma_record_etl"
+require_relative "hash_scan_based_etl_scanner"
+
+class Polyphemus
+  class MagmaRecordFileEtl < Polyphemus::MagmaRecordEtl
+    def scanner
+      HashScanBasedEtlScanner.new
+    end
+
+    def scanner_definition
+      scanner_def = super
+
+      scanner_def.start_batch_state do |cursor|
+        # Use query instead of retrieve so we don't have to
+        #   merge in a separate file md5 query later.
+        query_request = Etna::Clients::Magma::QueryRequest.new(
+          project_name: cursor[:project_name],
+          query: [cursor[:model_name]],
+          order: "updated_at",
+          page: 1,
+        )
+
+        prepare_request(cursor, query_request)
+
+        query_request
+      end.execute_batch_find do |query_request, i|
+        query_request.page_size = @limit * i
+
+        begin
+          self.magma_client.query(query_request).answer.map do |result|
+            { result.first => result.last.map.with_index { |value, i|
+              [queried_attribute_names[i], value]
+            }.to_h }
+          end
+        rescue Etna::Error => e
+          {} if e.message =~ /Page.*not found/
```

Comment:
This is the error thrown when there are no query results...feels like there should be a better way, but I don't think we can rely on status code.

Reply:
Seems like a bug... Shouldn't it return a 200 with an empty answer?

Reply:
Hm, let me investigate. I thought the Timur UI throws a 422 on the Search page when there are no results, so I assumed that was the same behavior for `/query`

Reply:
Oh, so when you page a query or retrieve request, and you include both `page` and `page_size` options, `Magma::Question` will throw a ["Page X not found" exception with a 422](https://github.com/mountetna/monoetna/blob/master/magma/lib/magma/query/question.rb#L307) when the specified doesn't exist. So this is the correct behavior within an ETL context.